rpc/block_svc_routines.c

```diff
@@ -5032,7 +5034,7 @@ block_list_cli_1_svc_st(blockListCli *blk, struct svc_req *rqstp)
         GB_METADIR, blk->volume, strerror(errno));
   }
 
-  if (errCode < 0) {
+  if (errCode != 0) {
```

Comment:
@amarts This was intentionally set to '< 0'
Some of the functions might set errCode to -1, if that is the case just reset it to GB_DEFAULT_ERRCODE (because errCode shouldn't be negative), Else carry forward the actual errCode returned, like ENOMEM, EIO, ENOENT etc